tests/sentry/snuba/metrics/test_query.py

```diff
@@ -219,6 +206,184 @@ def test_validate_order_by_field_in_select():
+    The example should fail because session crash free rate is generated from
+    counters entity while p50 of duration will go to distribution
+    """
+    metric_field_1 = MetricField(op=None, metric_name=SessionMetricKey.CRASH_FREE_RATE.value)
+    metric_field_2 = MetricField(op="p50", metric_name=TransactionMetricKey.DURATION.value)
+    metrics_query_dict = (
+        MetricsQueryBuilder()
+        .with_select([metric_field_1, metric_field_2])
+        .with_orderby(
+            [
+                OrderBy(field=metric_field_1, direction=Direction.ASC),
+                OrderBy(field=metric_field_2, direction=Direction.ASC),
+            ]
+        )
+        .to_metrics_query_dict()
+    )
+
+    # Test that ensures an instance of `InvalidParams` is raised when requesting an orderBy field
+    # that is not present in the select
+    with pytest.raises(
+        InvalidParams, match="Selected 'orderBy' columns must belongs to the same entity"
+    ):
+        MetricsQuery(**metrics_query_dict)
+
+
+def test_validate_many_order_by_fields_in_select_fails_diff_metrics_2():
+    """
+    This example should fail because session crash free rate is generated from
+    counters while session user crash free rate is generated from sets
+    """
+    metric_field_1 = MetricField(op=None, metric_name=SessionMetricKey.CRASH_FREE_RATE.value)
+    metric_field_2 = MetricField(op=None, metric_name=SessionMetricKey.CRASH_FREE_USER_RATE.value)
+    metrics_query_dict = (
+        MetricsQueryBuilder()
+        .with_select([metric_field_1, metric_field_2])
+        .with_orderby(
+            [
+                OrderBy(field=metric_field_1, direction=Direction.ASC),
+                OrderBy(field=metric_field_2, direction=Direction.ASC),
+            ]
+        )
+        .to_metrics_query_dict()
+    )
+
+    # Test that ensures an instance of `InvalidParams` is raised when requesting an orderBy field
+    # that is not present in the select
+    with pytest.raises(
+        InvalidParams, match="Selected 'orderBy' columns must belongs to the same entity"
+    ):
+        MetricsQuery(**metrics_query_dict)
+
+
+def test_validate_many_order_by_fields_in_select_success():
+    """
+    This example should pass because both session crash free rate
+    and sum(session) both go to the entity counters
+    """
+    metric_field_1 = MetricField(op=None, metric_name=SessionMetricKey.CRASH_FREE_RATE.value)
+    metric_field_2 = MetricField(op="sum", metric_name=SessionMetricKey.DURATION.value)
+    metrics_query_dict = (
+        MetricsQueryBuilder()
+        .with_select([metric_field_1, metric_field_2])
+        .with_orderby(
+            [
+                OrderBy(field=metric_field_1, direction=Direction.ASC),
+                OrderBy(field=metric_field_2, direction=Direction.ASC),
+            ]
+        )
+        .to_metrics_query_dict()
+    )
+
+    MetricsQuery(**metrics_query_dict)
+
+
+def test_validate_many_order_by_fields_are_in_select():
```

Comment:
This test is redundant as it pretty much does the same thing as https://github.com/getsentry/sentry/pull/35955/files#diff-21ea3024bbaf61c98c96daf5bcbefa0cad7ef6333a62ff5dc1ae1760030b042bR289-R303 

I'd just combine the tests honestly